ELKS not compiling when UNIX=y

The ELKS Stack does not compile when unix sockets are enabled. This is a regression. v0.6.0 does compile properly. The break happened this week. Suspected 12th, 13th, 14th or 15th commits.

**Configuration**
- Version: Head (16th Jun 2022)
- Emulator or HW platform: Compilation Issue

- Does the problem always happen ?
The problem is reproducible
- List the steps to reproduce the problem.

- Screenshot(s):
- Log file:

```
unix/af_unix.c:337:6: error: 'current' undeclared (first use in this function)
  if (current->signal /* & ~current->blocked */ )
      ^~~~~~~
unix/af_unix.c:346:5: warning: implicit declaration of function 'down' [-Wimplicit-function-declaration]
     down(&upd->sem);
     ^~~~
unix/af_unix.c:352:6: warning: implicit declaration of function 'send_sig' [-Wimplicit-function-declaration]
      send_sig(SIGKILL, current, 1);
      ^~~~~~~~
unix/af_unix.c:352:15: error: 'SIGKILL' undeclared (first use in this function)
      send_sig(SIGKILL, current, 1);
               ^~~~~~~
unix/af_unix.c:376:5: warning: implicit declaration of function 'up' [-Wimplicit-function-declaration]
     up(&upd->sem);
     ^~
unix/af_unix.c: In function 'unix_write':
unix/af_unix.c:391:15: error: 'SIGPIPE' undeclared (first use in this function)
      send_sig(SIGPIPE, current, 1);
               ^~~~~~~
unix/af_unix.c:391:24: error: 'current' undeclared (first use in this function)
      send_sig(SIGPIPE, current, 1);
                        ^~~~~~~
unix/af_unix.c:428:15: error: 'SIGKILL' undeclared (first use in this function)
      send_sig(SIGKILL, current, 1);
               ^~~~~~~
unix/af_unix.c: In function 'unix_select':
unix/af_unix.c:483:6: warning: implicit declaration of function 'select_wait' [-Wimplicit-function-declaration]
      select_wait(sock->wait);
      ^~~~~~~~~~~
```

**Additional information**
Unix sockets disabled is the workaround.

--- a/elks/Makefile
+++ b/elks/Makefile
@@ -124,8 +124,8 @@ net/net.a:
 
 tools:
 	${MAKE} -C tools all
-	-rm include/linuxmt/compiler-generated.h
-	-rm kernel/version.o
+	-rm -f include/linuxmt/compiler-generated.h
+	-rm -f kernel/version.o
 
 #########################################################################
 # Compiler-generated definitions not given as command arguments.

--- a/elks/include/linuxmt/socket.h
+++ b/elks/include/linuxmt/socket.h
@@ -54,8 +54,10 @@ struct msghdr {
 
 #ifdef __KERNEL__
 struct proto_ops;
-extern int sock_register(int,struct proto_ops *);
-extern int move_addr_to_user(char *,size_t,char *,int *);
+struct socket;
+int sock_register(int,struct proto_ops *);
+int move_addr_to_user(char *,size_t,char *,int *);
+int sock_awaitconn(struct socket *mysock, struct socket *servsock, int flags);
 #endif
 
 #endif

--- a/elks/net/nano/af_nano.c
+++ b/elks/net/nano/af_nano.c
@@ -11,14 +11,13 @@
 #include <linuxmt/config.h>
 #include <linuxmt/socket.h>
 #include <linuxmt/fs.h>
+#include <linuxmt/sched.h>
 #include <linuxmt/mm.h>
 #include <linuxmt/stat.h>
 #include <linuxmt/fcntl.h>
 #include "af_nano.h"
 
 #ifdef CONFIG_NANO
-
-int sock_awaitconn(register struct socket *mysock, struct socket *servsock, int flags);
 
 struct nano_proto_data nano_datas[NSOCKETS_NANO];
 

--- a/elks/net/unix/af_unix.c
+++ b/elks/net/unix/af_unix.c
@@ -11,6 +11,7 @@
 #include <linuxmt/config.h>
 #include <linuxmt/socket.h>
 #include <linuxmt/fs.h>
+#include <linuxmt/sched.h>
 #include <linuxmt/mm.h>
 #include <linuxmt/stat.h>
 #include <linuxmt/fcntl.h>
@@ -309,8 +310,8 @@ static int unix_getname(struct socket *sock,
     } else
 	upd = UN_DATA(sock);
 
-    return move_addr_to_user(&upd->sockaddr_un, upd->sockaddr_len,
-				    usockaddr, usockaddr_len);
+    return move_addr_to_user((char *)&upd->sockaddr_un, upd->sockaddr_len,
+				    (char *)usockaddr, usockaddr_len);
 }
 
 static int unix_read(struct socket *sock, char *ubuf, int size, int nonblock)

